returnn/tf/layers/signal_processing.py

```diff
@@ -607,6 +607,57 @@ def get_out_data_from_opts(cls, fft_size, use_rfft=True, nr_of_channels=1, **kwa
             .copy_template(dtype="complex64"))
 
 
+class StftLayer(_ConcatInputLayer):
+  """
+  A generic STFT layer.
+  """
+
+  layer_class = "stft"
+  recurrent = True  # we should not shuffle in the time-dimension
+
+  # noinspection PyUnusedLocal
+  def __init__(self, frame_shift, frame_size, fft_size=None, **kwargs):
+    fft_size = fft_size or frame_size
+    assert "n_out" not in kwargs
+    assert "out_type" not in kwargs
+    super(StftLayer, self).__init__(**kwargs)
+    assert not self.input_data.sparse
+    assert self.input_data.have_batch_axis()
+    assert self.input_data.have_time_axis()
+    out_dim = fft_size // 2 + 1
+    in_dim = self.input_data.feature_dim_or_sparse_dim
+    input_data = self.input_data.copy_as_batch_feature_major()
```

Comment:
Is this `copy_as_batch_feature_major` correct? I think you should handle it afterwards. See `PoolLayer` or `ConvLayer`.